storage/src/main/java/org/openmbee/sdvc/storage/MinIOStorage.java

```diff
@@ -0,0 +1,135 @@
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class MinIOStorage implements ArtifactStorage {
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    private MinioClient minio;
+
+    @Value("${minio.endpoint}")
+    private String ENDPOINT;
+
+    @Value("${minio.access_key}")
+    private String ACCESS_KEY;
+
+    @Value("${minio.secret_key}")
+    private String SECRET_KEY;
+
+    @Value("${minio.bucket:#{null}}")
+    private Optional<String> BUCKET;
+
+    @Value("${minio.region:#{null}}")
+    private Optional<String> REGION;
+
+    private MimeTypes mimeTypes = MimeTypes.getDefaultMimeTypes();
+
+    private MinioClient getClient()
+        throws IOException, InvalidKeyException, InvalidResponseException, RegionConflictException, InsufficientDataException, NoSuchAlgorithmException, ServerException, InternalException, XmlParserException, InvalidBucketNameException, ErrorResponseException {
+        if (minio == null) {
+            MinioClient.Builder builder = MinioClient.builder().endpoint(ENDPOINT).credentials(ACCESS_KEY, SECRET_KEY);
+            REGION.ifPresent(builder::region);
+            minio = builder.build();
+            boolean isExist = minio.bucketExists(BucketExistsArgs.builder().bucket(getBucket()).build());
+            if (!isExist) {
+                minio.makeBucket(MakeBucketArgs.builder().bucket(getBucket()).build());
+            }
+        }
+        return minio;
+    }
+
+    @Override
+    public byte[] get(String location, ElementJson element, String mimetype) {
+        GetObjectArgs goa = GetObjectArgs.builder()
+            .bucket(getBucket())
+            .object(location)
+            .build();
+        byte[] ba = null;
+        try {
+            ba = getClient().getObject(goa).readAllBytes();
+        } catch (Exception e) {
+            logger.error("Error getting object: ", e);
+        }
+        return ba;
+    }
+
+    @Override
+    public String store(byte[] data, ElementJson element, String mimetype) {
+        String location = buildLocation(element, mimetype);
+        PutObjectArgs poa = PutObjectArgs.builder()
+            .bucket(getBucket())
+            .object(location)
+            .contentType(mimetype)
+            .stream(new ByteArrayInputStream(data), -1,  1048576 * 5) // 5MB Part size
+            .build();
+
+        try {
+            getClient().putObject(poa);
+        } catch (Exception e) {
+            logger.error("Error putting object: ", e);
+        }
+        return location;
+    }
+
+    private String buildLocation(ElementJson element, String mimetype) {
+        int it = 1;
+        try {
+            Iterable<Result<Item>> results = getClient().listObjects(
+                ListObjectsArgs.builder().bucket(getBucket()).build());
+            for (Result<Item> result : results) {
+                it++;
```

Comment:
generating a datetime string's probably better - saves having to list and iterate all bucket items
it's not like the numbers mean anything since it's based on all artifacts across all projects